ktx-library-codelab/step-03/app/src/main/java/com/example/android/kotlinktxworkshop/MainActivity.kt

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.android.kotlinktxworkshop
+
+import android.Manifest
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.os.Looper
+import androidx.appcompat.app.AppCompatActivity
+import com.example.android.myktxlibrary.createLocationRequest
+import com.example.android.myktxlibrary.findAndSetText
+import com.example.android.myktxlibrary.hasPermission
+import com.example.android.myktxlibrary.showLocation
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationCallback
+import com.google.android.gms.location.LocationResult
+import com.google.android.gms.location.LocationServices
+
+class MainActivity : AppCompatActivity() {
+
+    private var listeningToUpdates = false
+
+    private val locationCallback: LocationCallback = object : LocationCallback() {
+        override fun onLocationResult(locationResult: LocationResult?) {
+            if (locationResult != null) {
+                showLocation(R.id.textView, locationResult.lastLocation)
+            }
+        }
+    }
+
+    private lateinit var fusedLocationClient: FusedLocationProviderClient
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(this)
+    }
+
+    override fun onStart() {
+        super.onStart()
+
+        if (!hasPermission(Manifest.permission.ACCESS_FINE_LOCATION)) {
+            requestPermissions(arrayOf(Manifest.permission.ACCESS_FINE_LOCATION), 0)
+        }
+
+        getLastKnownLocation()
+        startUpdatingLocation()
+    }
+
+    private fun getLastKnownLocation() {
+        fusedLocationClient.lastLocation.addOnSuccessListener { lastLocation ->
+            showLocation(R.id.textView, lastLocation)
+        }.addOnFailureListener { e ->
+            findAndSetText(R.id.textView, "Unable to get location.")
+            e.printStackTrace()
+        }
+    }
+
+    private fun startUpdatingLocation() {
+        fusedLocationClient.requestLocationUpdates(
+            createLocationRequest(),
+            locationCallback,
+            Looper.getMainLooper()
+        ).addOnSuccessListener { listeningToUpdates = true }
```

Comment:
Nit: indentation

Should we move `.addOnSuccessListener` to another line? `.addOnFailureListener` has more indentation and the combination of these two look weird